pkg/controllermanager/controller/shoot/shoot_control_delete.go

```diff
@@ -291,6 +291,16 @@ func (c *Controller) runDeleteShootFlow(o *operation.Operation) *gardencorev1alp
 			Fn:           flow.TaskFn(botanist.WaitUntilManagedResourcesDeleted).Timeout(10 * time.Minute),
 			Dependencies: flow.NewTaskIDs(deleteManagedResources),
 		})
+		destroyCloudSpecificControlPlane = g.Add(flow.Task{
```

Comment:
I think we should delete it after the kube-apiserver deployment is deleted. It's relevant for the kube-apiserver exposure, and if it is gone then we can safely delete all exposure-relevant components.